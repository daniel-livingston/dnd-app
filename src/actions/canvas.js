export const SAVE_CANVAS = "SAVE_CANVAS";
export const saveCanvas = (canvas) => ({
	type: SAVE_CANVAS,
	canvas: canvas,
});

export const SET_BRUSH_COLOR = "SET_BRUSH_COLOR";
export const setBrushColor = (brushColor) => ({
	type: SET_BRUSH_COLOR,
	brushColor,
});

export const SET_IS_DRAWING = "SET_IS_DRAWING";
export const setIsDrawing = (isDrawing) => ({
	type: SET_IS_DRAWING,
	isDrawing,
});

export const SET_LINE_WIDTH = "SET_LINE_WIDTH";
export const setLineWidth = (lineWidth) => ({
	type: SET_LINE_WIDTH,
	lineWidth,
});
