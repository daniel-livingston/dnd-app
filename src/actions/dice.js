export const ROLL_DICE = "ROLL_DICE";
export const rollDice = (roll) => ({
	type: ROLL_DICE,
	roll,
});

export const UPDATE_DICE_NUMBER = "UPDATE_DICE_NUMBER";
export const updateDiceNumber = (number) => ({
	type: UPDATE_DICE_NUMBER,
	number,
});

export const UPDATE_DICE_SIDES = "UPDATE_DICE_SIDES";
export const updateDiceSides = (sides) => ({
	type: UPDATE_DICE_SIDES,
	sides,
});

export const UPDATE_DICE_MODIFIER = "UPDATE_DICE_MODIFIER";
export const updateDiceModifier = (modifier) => ({
	type: UPDATE_DICE_MODIFIER,
	modifier,
});

export const SET_HISTORY = "SET_HISTORY";
export const setHistory = (history) => ({
	type: SET_HISTORY,
	history,
});

export const ADD_TO_HISTORY = "ADD_TO_HISTORY";
export const addToHistory = (roll) => ({
	type: ADD_TO_HISTORY,
	roll,
});
