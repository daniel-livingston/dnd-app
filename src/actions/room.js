export const JOIN_ROOM = "JOIN_ROOM";
export const joinRoom = (roomData) => ({
	type: JOIN_ROOM,
	data: roomData,
});

export const LEAVE_ROOM = "LEAVE_ROOM";
export const leaveRoom = () => ({
	type: LEAVE_ROOM,
});
