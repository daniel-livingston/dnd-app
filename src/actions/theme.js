const SET_THEME = "SET_THEME";
const setTheme = (theme) => ({
	type: SET_THEME,
	theme,
});

export { SET_THEME, setTheme };
