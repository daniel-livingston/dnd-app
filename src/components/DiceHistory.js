import React from "react";
import { connect } from "react-redux";

export const DiceHistory = (props) => (
	<div className='dice-history'>
		{props.history.map((data) => (
			<div key={`${data.time.valueOf()}`} className='dice-history__item'>
				<div className='dice-history__item__overview'>
					{data.time.toLocaleTimeString()}: <span className='accent'>{data.name}</span>{" "}
					rolled <span className='accent'>{data.total}</span>!
				</div>
				<div className='dice-history__item__details'>
					{data.number}d{data.sides}: {data.rolls.join(", ")} +{data.modifier}
				</div>
			</div>
		))}
	</div>
);

const mapStateToProps = (state) => ({
	history: state.dice.history,
});

export default connect(mapStateToProps)(DiceHistory);
