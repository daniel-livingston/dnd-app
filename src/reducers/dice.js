import {
	ADD_TO_HISTORY,
	ROLL_DICE,
	SET_HISTORY,
	UPDATE_DICE_NUMBER,
	UPDATE_DICE_SIDES,
	UPDATE_DICE_MODIFIER,
} from "../actions/dice";

export const defaultState = {
	number: 1,
	sides: 20,
	modifier: 0,
	last: {},
	history: [],
};

const diceReducer = (state = defaultState, action) => {
	switch (action.type) {
		case ADD_TO_HISTORY: {
			return {
				...state,
				history: [action.roll].concat(state.history),
			};
		}
		case ROLL_DICE: {
			return {
				...state,
				last: action.roll,
				history: [action.roll].concat(state.history),
			};
		}
		case SET_HISTORY: {
			return {
				...state,
				history: action.history,
			};
		}
		case UPDATE_DICE_NUMBER: {
			return {
				...state,
				number: action.number,
			};
		}
		case UPDATE_DICE_SIDES: {
			return {
				...state,
				sides: action.sides,
			};
		}
		case UPDATE_DICE_MODIFIER: {
			return {
				...state,
				modifier: action.modifier,
			};
		}
		default: {
			return state;
		}
	}
};

export default diceReducer;
