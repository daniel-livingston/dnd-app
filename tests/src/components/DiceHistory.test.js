import React from "react";
import { Provider } from "react-redux";
import { mount, shallow } from "enzyme";
import store from "../../../src/store/configureStore";
import DiceHistory from "../../../src/components/DiceHistory";
import { rollDice } from "../../../src/actions/dice";

describe("<DiceHistory>", () => {
	test("renders correctly with empty history", () => {
		const wrapper = mount(
			<Provider store={store}>
				<DiceHistory />
			</Provider>
		);
		expect(wrapper).toMatchSnapshot();
	});

	test("renders correctly with populated history", () => {
		store.dispatch(
			rollDice({
				rolls: [1, 3, 4],
				number: 3,
				sides: 6,
				modifier: 0,
				total: 8,
				time: new Date(0),
				name: "Testname",
			})
		);
		const wrapper = mount(
			<Provider store={store}>
				<DiceHistory />
			</Provider>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
