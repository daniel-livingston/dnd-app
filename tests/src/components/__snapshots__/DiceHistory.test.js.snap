// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiceHistory> renders correctly with empty history 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(DiceHistory)>
    <DiceHistory
      dispatch={[Function]}
      history={Array []}
    >
      <div
        className="DiceHistory"
      />
    </DiceHistory>
  </Connect(DiceHistory)>
</Provider>
`;

exports[`<DiceHistory> renders correctly with populated history 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(DiceHistory)>
    <DiceHistory
      dispatch={[Function]}
      history={
        Array [
          Object {
            "modifier": 0,
            "name": "Testname",
            "number": 3,
            "rolls": Array [
              1,
              3,
              4,
            ],
            "sides": 6,
            "time": 1970-01-01T00:00:00.000Z,
            "total": 8,
          },
        ]
      }
    >
      <div
        className="DiceHistory"
      >
        <DiceHistoryItem
          key="Testname:0"
          modifier={0}
          name="Testname"
          number={3}
          rolls={
            Array [
              1,
              3,
              4,
            ]
          }
          sides={6}
          time={1970-01-01T00:00:00.000Z}
          total={8}
        >
          <div
            className="DiceHistoryItem"
          >
            <div
              className="DiceHistoryItem-overview"
            >
              7:00:00 PM
              : 
              <span
                className="accent"
              >
                Testname
              </span>
               rolled
               
              <span
                className="accent"
              >
                8
              </span>
            </div>
            <div
              className="DiceHistoryItem-details"
            >
              3
              d
              6
              : 
              1, 3, 4
               + 
              0
            </div>
          </div>
        </DiceHistoryItem>
      </div>
    </DiceHistory>
  </Connect(DiceHistory)>
</Provider>
`;
